cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -1519,7 +1519,6 @@ public MNode getMNode(MNode deviceMNode, String measurementName) {
     // do not use limit and offset in sub-queries unless offset is 0, otherwise the results are
     // not combinable
     if (offset != 0) {
-      plan.setLimit(0);
```

Comment:
This should not be removed. Consider this case: 
The offset is 10000, the limit is 10, and the results are distributed on 3 nodes. If you keep the limit, each of the 3 nodes will only return 10 results, 30 results in total, which will all be filtered when the offset is applied.
You may set the limit to `limit + offset` so the unnecessary fetch can be reduced.